resin/logs.py

```diff
@@ -1,133 +1,194 @@
-from pubnub import Pubnub
 from functools import wraps
-import logging
+import json
+try:  # Python 3 imports
+    from urllib.parse import urljoin
+except ImportError:  # Python 2 imports
+    from urlparse import urljoin
+
+from twisted.internet import reactor, ssl
+from twisted.internet.defer import Deferred
+from twisted.internet.protocol import Protocol
+from twisted.web.client import Agent
+from twisted.web.http_headers import Headers
+from twisted.internet.ssl import ClientContextFactory
+
+from threading import Thread
 
 from .base_request import BaseRequest
+from . import exceptions
 from .models.config import Config
 from .models.device import Device
+from .settings import Settings
 
 
-logging.basicConfig(format='%(levelname)s:%(message)s')
+class WebClientContextFactory(ClientContextFactory):
+    """
+    This is low level class and is not meant to be used by end users directly.
+    """
 
+    def getContext(self, hostname, port):
+        return ClientContextFactory.getContext(self)
 
-def _print_deprecation_warning():
-    logging.warning(" using legacy logging services, this will stop working shortly.\nPlease update to ensure logs are correctly retrieved in future.")
 
+class StreamingParser(Protocol):
+    """
+    This is low level class and is not meant to be used by end users directly.
+    """
 
-# TODO: https://github.com/resin-io/resin-sdk/pull/277/files
-class Logs(object):
+    def __init__(self, callback, error):
+        self.callback = callback
+        self.error = error
+
+    def dataReceived(self, data):
+        d = Deferred()
```

Comment:
Why do you construct a Deferred here? You can process the data immediately

Reply:
Yeah, I tidied up the code here and it's easier to read now.

Reply:
You're still using a Deferred for no reason. Just call the error callback directly